### PR TITLE
test(blend): extensive unit tests for core->core logic

### PR DESCRIPTION
**Pull request by ntn-x2**
## 1. What does this PR implement?

PR 1/3. This setups up the testing infrastructure and introduces unit test cases for core->core behaviour. Core->edge as well as the swarm logic (e.g., peer re-dials) will be introduced in subsequent PRs. We cover all edge cases and potential concurrency issues arising from how rust-lip2p works.

## 2. Does the code have enough context to be clearly understood?

Yes.

## 3. Who are the specification authors and who is accountable for this PR?

@ntn-x2 

## 4. Is the specification accurate and complete?

Yes.

## 5. Does the implementation introduce changes in the specification?

No.

## Checklist

* [x] 1. The PR title follows the Conventional Commits [specification](https://www.notion.so/How-to-open-a-Pull-Request-215261aa09df80deb538ed59f5e4e0b5).
* [x] 2. Description added.
* [x] 3. Context and links to Specification document(s) added.
* [x] 4. Main contact(s) (developers and specification authors) added
* [x] 5. Implementation and Specification are 100% in sync including changes. This is critical.
* [x] 6. Link PR to a specific milestone.

### CODE REVIEW

File: nomos-blend/network/src/core/with_core/behaviour/tests/utils.rs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    EncapsulationInputs::new(
+        repeat_with(Ed25519PrivateKey::generate)
+            .take(3)
+            .map(|recipient_signing_key| {
+                let recipient_signing_pubkey = recipient_signing_key.public_key();
+                EncapsulationInput::new(
+                    Ed25519PrivateKey::generate(),
+                    &recipient_signing_pubkey,
+                    ProofOfQuota::from([0u8; PROOF_OF_QUOTA_SIZE]),
+                    ProofOfSelection::dummy(),
```
Comment (youngjoon-lee) [inline nomos-blend/network/src/core/with_core/behaviour/tests/utils.rs]:
```suggestion
                    ProofOfQuota::dummy(),
                    ProofOfSelection::dummy(),
```
It seems that we can use `dummy()` method for `ProofOfQuota` as well? All tests pass.

Comment (ntn-x2) [reply nomos-blend/network/src/core/with_core/behaviour/tests/utils.rs]: Yup, good catch! https://github.com/logos-co/nomos/pull/1542/commits/d4428faf585461db71606973db81d09614acad75
